server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/WebAdminServer.java

```diff
@@ -174,7 +169,6 @@ public void destroy() {
     }
 
     public Port getPort() {
-        Preconditions.checkState(port != null, "WebAdminServer should be configured.");
-        return port;
+        return new Port(service.port());
```

Comment:
I don't get it, you assume getPort() will be call only once on a given started webAdminServer?

Reply:
Well, if the server is not started the port is not allocated, but we now delegate port management to SperkService- which have such preconditions built in - so it is no more necessary to do it in our code base.

Reply:
oh sorry I misread, nice